### PR TITLE
Bsc4 2

### PR DESCRIPTION
**Pull request by etiennesky**
this merge requests fixes some minor issues and adds support for "reduced" output which do not contains variables for clear-sky fluxes. missing fields in ECMean output are set as N/A.

### CODE REVIEW

File: ECmean/EC-mean.sh [hunk 1]
```diff
@@ -107,6 +107,17 @@ do_ocean=0
     echo "*II* ecmean accounts for nemo output"
 export do_ocean
 
+# -- other options
+
+# process 3D vars (most of which which are in SH files) ? 
+# set to 0 if you only want simple diags e.g. Gregory plots or if using the reduced outclass
+[[ -z "${ECE3_POSTPROC_ECM_3D_VARS:-}" ]] && do_3d_vars=1 || do_3d_vars=${ECE3_POSTPROC_ECM_3D_VARS}
```
Comment (plesager) [inline ECmean/EC-mean.sh]: These lines (110-115) should be in the config file of the platform that needs it. Others can keep the original, no need for extra variable. This is exactly like:
[[[ -z ${ECE3_POSTPROC_POSTDIR:-} ]] && export ECE3_POSTPROC_POSTDIR='$SCRATCH/ECEARTH-RUNS/${EXPID}/post'
we do not want to spread configs all over the place, better keep it in the config file.

**Review by plesager — CHANGES_REQUESTED** [review]

File: ECmean/EC-mean.sh [hunk 2]
```diff
@@ -107,6 +107,17 @@ do_ocean=0
...
+export do_3d_vars
+
+# compute clear sky fluxes, set to 0 if using the reduced outclass
+[[ -z "${ECE3_POSTPROC_ECM_CLEAR_FLUX:-}" ]] && do_clear_flux=1 || do_clear_flux=${ECE3_POSTPROC_ECM_CLEAR_FLUX}
+export do_clear_flux
```
Comment (plesager) [inline ECmean/EC-mean.sh]: Same here

**Comment by plesager** [comment]
I am also concerned with the reduced output cases. If a file is missing (but shouldn't), the code will continue: we have to be clear that is ok to have that file missing. Something along those lines in pseudo code:
```
 if file missing then
  if not do_3d_var then OK else raise error
```

File: ECmean/EC-mean.sh (hunk 1 above)
Comment (etiennesky) [reply ECmean/EC-mean.sh]: Hi @plesage my change is comparable to what is already done in hiresclim2/master_hiresclim.sh, but if you preffer to put them in the config files that is ok with me.

```
# IFS monthly [ON by default]
# IFS daily and 6hrs flag for u,v,t,z 3d field + tas, totp extraction [OFF by default]
ifs_monthly=${ECE3_POSTPROC_HC_IFS_MONTHLY:-1}
ifs_monthly_mma=${ECE3_POSTPROC_HC_IFS_MONTHLY_MMA:-0}
ifs_daily=${ECE3_POSTPROC_HC_IFS_DAILY:-0}
ifs_6hrs=${ECE3_POSTPROC_HC_IFS_6HRS:-0}

```

**Comment by etiennesky** [comment]
Hi @plesager about the reduced output, the missing files are usually not very important for core ECMean diags (such as sea ice in atm-only simulation, which IMHO does not make any sense at all!). 

I can adapt the scripts to check for  do_clear_flux and do_3d and see what is left.

File: ECmean/EC-mean.sh (hunk 1 above)
Comment (plesager) [reply ECmean/EC-mean.sh]: Got your point, although these ECE3_POSTPROC_HC_IFS_* variables are in the config files, commented. So no need to dive in the code to find out that they are available and can be set/unset.

**Comment by plesager** [comment]
About the reduced output: ok, we just need to be sure that they are not needed for ecmean.

File: ECmean/EC-mean.sh (hunk 1 above)
Comment (etiennesky) [reply ECmean/EC-mean.sh]: so if I add them to the config files, commented, it would be ok?

Comment (plesager) [reply ECmean/EC-mean.sh]: yep

**Comment by etiennesky** [comment]
Hi @plesager I have cleaned up the do_clear_flux and do_3d_vars code added them to the config files.

About the reduced output, I have tried to keep changes at minimum but this is the easiest I can see it. If a file should exists and doesn't, global_mean.sh will ignore it and if the files are required for diagnostics then ECMean will complain.

If you really want to I can replace do_clear_flux with reduced_output and remove some variables from the varlists insteadof the various [ -f ] tests.

**Pull request merged**